Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,356 @@
+        combobox.removeAttr('aria-labelledby');
+
+        if (combobox.attr('data-label-id-index') !== undefined)
+        {
+          textbox.attr('data-label-id-index', combobox.attr('data-label-id-index'));
+          combobox.removeAttr('data-label-id-index');
+        }
+      }
+      else
+      {
+        textbox.attr('aria-labelledby', internetExplorerScreenReaderLabelID);
+      }
+
+      if (combobox.attr('aria-describedby') !== undefined)
+      {
+        textbox.attr('aria-describedby', combobox.attr('aria-describedby'));
+        combobox.removeAttr('aria-describedby');
+      }
+
+      combobox = textbox;
+    }
+
+    const ClearError = () =>
+    {
+      if (textbox.hasClass('error'))
+      {
+        textbox.removeClass('error');
+        textbox.trigger('hideInformationPopUp');
+      }
+    }
+  
+    const SetError = (message: string) =>
+    {
+      textbox.addClass('error');
+      textbox.trigger('showInformationPopUp', { 'message': message });
+    }
+
+    const errorHandler = (error: Sys.Net.WebServiceError) =>
+    {
+      SetError(resources.LoadIconFailedErrorMessage);
+    };
+
+    textbox.autocomplete(this._controlServiceUrl, 'Search', 'SearchExact',
+      {
+        extraParams: searchContext,
+        isAutoPostBackEnabled: this._isAutoPostBackEnabled,
+        nullValue: this._nullValueString, // the hidden field value indicating that no value has been selected
+        searchStringValidationParams:
+        {
+          inputRegex: new RegExp(searchStringValidationInfo.ValidSearchStringRegex),
+          dropDownTriggerRegex: new RegExp(searchStringValidationInfo.ValidSearchStringForDropDownRegex),
+          dropDownTriggerRegexFailedMessage: searchStringValidationInfo.SearchStringForDropDownDoesNotMatchRegexMessage,
+          getDropDownSearchString: (searchString: string) =>
+          {
+            return searchStringValidationInfo.IgnoreSearchStringForDropDownUponValidInput ? this.GetDropDownSearchStringForValidInput(searchString) : searchString;
+          }
+        },
+        max: completionSetCount, // Set query limit
+
+        dropDownDisplayDelay: dropDownDisplayDelay,
+        dropDownRefreshDelay: dropDownRefreshDelay,
+        selectionUpdateDelay: selectionUpdateDelay,
+
+        noDataFoundMessage: resources.NoDataFoundMessage,
+        autoFill: true,
+        matchContains: true,
+        combobox: combobox,
+        selectListID: this._selectListID,
+        informationPopUpID: this._informationPopUpID,
+        dropDownButtonID: button.attr('id'),
+        inputAreaClass: 'content',
+        // this can be set to true/removed once the problem is fixed that an empty textbox still selects the first element, making it impossible to clear the selection
+        selectFirst: function (inputValue: string, searchTerm: string)
+        {
+          return inputValue.length > 0;
+        },
+        dataType: 'json',
+        parse: function (data: any)
```

Comment:
data can be assigned a type. I think it's an array of `row`, and row is of type BocAutoCompleteReferenceValue_Item